aten/src/ATen/templates/TensorBody.h

```diff
@@ -487,7 +497,11 @@ class TORCH_API Tensor {
 
   /// Returns the `TensorOptions` corresponding to this `Tensor`. Defined in
   /// TensorOptions.h.
-  TensorOptions options() const;
+  TensorOptions options() const {
```

Comment:
I could also move all of these methods back into `TensorMethods.cpp` like I mentioned in the description, and keep that file around. These all just seemed small enough to inline though

Reply:
This will also probably be a perf boost, I have some local benchmarks showing that options() function-call overhead is noticeable